--- o.qdbm-1.8.48/perl/Makefile.in	2006-01-17 09:25:38.000000000 +0100
+++ n.qdbm-1.8.48/perl/Makefile.in	2006-04-16 09:34:39.000000000 +0200
@@ -41,15 +41,15 @@
 
 all :
 	cd depot && [ -f Makefile ] || \
-	  $(RUNENV) perl Makefile.PL CC="$(CC)" INC="$(INC)" OPTIMIZE="$(OPTIMIZE)" \
+	  $(RUNENV) perl Makefile.PL CC="$(CC)" INC="$(INC)" OPTIMIZE="$(OPTIMIZE)" INSTALLDIRS=vendor \
 	    LD="$(LD)" LIBS="$(LIBS)" PREFIX="$(prefix)" ; \
 	  $(RUNENV) make
 	cd curia && [ -f Makefile ] || \
-	  $(RUNENV) perl Makefile.PL CC="$(CC)" INC="$(INC)" OPTIMIZE="$(OPTIMIZE)" \
+	  $(RUNENV) perl Makefile.PL CC="$(CC)" INC="$(INC)" OPTIMIZE="$(OPTIMIZE)" INSTALLDIRS=vendor \
 	    LD="$(LD)" LIBS="$(LIBS)" PREFIX="$(prefix)" ; \
 	  $(RUNENV) make
 	cd villa && [ -f Makefile ] || \
-	  $(RUNENV) perl Makefile.PL CC="$(CC)" INC="$(INC)" OPTIMIZE="$(OPTIMIZE)" \
+	  $(RUNENV) perl Makefile.PL CC="$(CC)" INC="$(INC)" OPTIMIZE="$(OPTIMIZE)" INSTALLDIRS=vendor \
 	    LD="$(LD)" LIBS="$(LIBS)" PREFIX="$(prefix)" ; \
 	  $(RUNENV) make
 	@printf '\n'
@@ -69,9 +69,9 @@
 
 
 install :
-	cd depot && $(RUNENV) make install
-	cd curia && $(RUNENV) make install
-	cd villa && $(RUNENV) make install
+	cd depot && $(RUNENV) make pure_install
+	cd curia && $(RUNENV) make pure_install
+	cd villa && $(RUNENV) make pure_install
 	mkdir -p $(DESTDIR)$(MYBINDIR)
 	cp -f $(MYBINS) $(DESTDIR)$(MYBINDIR)
 	mkdir -p $(DESTDIR)$(MYDATADIR)
